### PR TITLE
Prevent large fee txs

### PR DESCRIPTION
**Pull request by cgcardona**
* prevent Goose Egg txs

### CODE REVIEW

File: src/apis/avm/api.ts [hunk 1]
```diff
@@ -671,7 +718,50 @@ class AVMAPI extends JRPCAPI {
      *
      * @returns A signed transaction of type [[Tx]]
      */
-  signTx = (utx:UnsignedTx):Tx => this.keychain.signTx(utx);
+  signTx = (utx:UnsignedTx):Tx => {
+    let onlyAVAXFee:boolean = true;
+    const tx:BaseTx = utx.getTransaction();
+    const avaxAssetID:string = "n8XH5JY1EX5VYqDeAhB4Zd4GKxi9UNQy6oPpMsCAj1Q6xkiiL";
```
Comment (cgcardona) [inline src/apis/avm/api.ts:724]: Don't hardcode avax asset id

File: src/apis/avm/api.ts [hunk 2]
```diff
@@ -546,6 +548,51 @@ class AVMAPI extends JRPCAPI {
...
+      const output:AmountOutput = outs[i].getOutput() as AmountOutput; 
+
+      // only check secpoutputs
+      if(output.getOutputID() === AVMConstants.SECPOUTPUTID) {
+        outputTotal = outputTotal.add(output.getAmount());
+      }
+    }
+
+    const fee:BN = inputTotal.sub(outputTotal);
+    if((outputTotal.lte(new BN(10000000)) && fee.lte(outputTotal.div(new BN(2)))) || fee.lte(outputTotal.div(new BN(10000)))) {
```
Comment (collincusce) [inline src/apis/avm/api.ts:587]: Let's make these numbers go into AVMConstants.

Comment (cgcardona) [inline src/apis/avm/api.ts:587]: Make these numbers constants

File: src/apis/avm/api.ts [hunk 3]
```diff
@@ -546,6 +548,51 @@ class AVMAPI extends JRPCAPI {
...
+     *
+     * @remarks
+     * A "Goose Egg Transaction" is when the fee far exceeds a reasonable amount
+     */
+  checkGooseEgg = (utx:UnsignedTx): boolean => {
+    let inputTotal:BN = new BN(0);
+    let outputTotal:BN = new BN(0);
+    const tx:BaseTx = utx.getTransaction();
+
+    const ins:Array<TransferableInput> = tx.getIns();
```
Comment (collincusce) [inline src/apis/avm/api.ts:566]: Let's put a getSpend(assetID) on the UnsignedTx class.

File: src/apis/avm/api.ts [hunk 4]
```diff
@@ -546,6 +548,51 @@ class AVMAPI extends JRPCAPI {
...
+    for(let i:number = 0; i < ins.length; i++){
+      const input:AmountInput = ins[i].getInput() as AmountInput; 
+
+      // only check secpinputs
+      if(input.getInputID() === AVMConstants.SECPINPUTID) {
+        inputTotal = inputTotal.add(input.getAmount());
+      }
+    }
+
+    const outs:Array<TransferableOutput> = tx.getOuts();
```
Comment (collincusce) [inline src/apis/avm/api.ts:576]: Let's make a getBurned(assetID) on the UnsignedTx class.

File: src/apis/avm/api.ts (hunk 1 above)
Comment (collincusce) [inline src/apis/avm/api.ts:724]: this.getAVAAssetID()

**Pull request merged**